exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/AbstractParquetGroupScan.java

```diff
@@ -261,13 +262,17 @@ public GroupScan applyFilter(LogicalExpression filterExpr, UdfUtilities udfUtili
           return null;
         }
       }
-
-      if (ParquetRGFilterEvaluator.canDrop(filterPredicate, columnStatisticsMap, rowGroup.getRowCount())) {
+      rowGroup.setRowsMatch(ParquetRGFilterEvaluator.matches(filterPredicate, columnStatisticsMap, rowGroup.getRowCount()));
+      if (rowGroup.getRowsMatch() == ROWS_MATCH.NONE) {// No row comply to the filter => drop the row group
         continue;
       }
+      // DRILL_6259 : in case of array, we can't know if the real value of a repeated column exists until it is really filtered, hence we can't prune the filter
+      if (rowGroup.getRowsMatch() == ROWS_MATCH.ALL && statCollector.isRepeated(schemaPathsInExpr)) {
```

Comment:
`statCollector.isRepeated(schemaPathsInExpr))` -> I am not sure this is right place to do the check. Previously all was done during predicate evaluation, I think moving this here is quite correct. Please re-consider.

Reply:
Done